### PR TITLE
Support for Jupiter Swap API

### LINKED ISSUE #1: Add support for Jupiter SWAP API
The jupiter swap api has a function where they generate a transaction for you. Add a ping-thing version for the Jupiter swap api that basically retrieves a transaction body from the Jupiter swap api, then submits that transaction and follows up on its confirmation.

### PR DESCRIPTION
**Pull request by WilfredAlmeida**
Fixes #1

### CODE REVIEW

File: ping-thing-client-jupiter.mjs [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+      }
+    }
+
+    await sleep(5000);
+  }
+}
+
+// Record new slot on `firstShredReceived`
+const gSlotSent = { value: null, updated_at: 0 };
+async function watchSlotSent() {
```
Comment (linuskendall) [inline ping-thing-client-jupiter.mjs]: probably some of these should be moved to a separate library file if we're refactoring this

Comment (linuskendall) [reply ping-thing-client-jupiter.mjs]: (not a blocker for merging this PR, but maybe good to add as a new issue)

**Review by linuskendall — APPROVED** [review]
This looks good to me!

File: ping-thing-client-jupiter.mjs (hunk 1 above)
Comment (WilfredAlmeida) [reply ping-thing-client-jupiter.mjs]: They could be. However, I'm unaware of the deployment environment of the scripts, so I don't want to cause any dependency issues for standalone scripts. There's duplicate code that can be abstracted out.

**Comment by fanatid** [comment]
lgtm

as Linus mentioned we need to refactor scripts and move duplicated code to shared file
maybe it would be good to support args, like `ping-thing.mjs --jup sol-usdc` / `ping-thing.mjs --transfer sol-sol`

**Comment by WilfredAlmeida** [comment]
I'll refactor to abstract out the duplication and push.

> maybe it would be good to support args, like ping-thing.mjs --jup sol-usdc / ping-thing.mjs --transfer sol-sol

A CLI-like approach can be taken to this, however, I'd like to know if its a priority. We need to assess the reliability of this and test & deploy it as well. So I'd prefer to focus on deployment & testing first and then implement this.

**Pull request merged**